doc/api/child_process.md

```diff
@@ -27,10 +27,15 @@ ls.on('close', (code) => {
 ```
 
 By default, pipes for `stdin`, `stdout`, and `stderr` are established between
-the parent Node.js process and the spawned child. It is possible to stream data
-through these pipes in a non-blocking way. *Note, however, that some programs
-use line-buffered I/O internally. While that does not affect Node.js, it can
-mean that data sent to the child process may not be immediately consumed.*
+the parent Node.js process and the spawned child. These pipes have
+limited (and platform-specific) capacity, and if a process writes stdout
+in excess of that limit without capturing the output, that can cause the process
+to hang. This is identical behavior to the behavior of pipes in the shell.
```

Comment:
Is "hang" a term we want to use? (We may already use it elsewhere, but I'm not sure it is sufficiently specific. I'm also not sure what a better term is, though.  /cc @nodejs/documentation )

Reply:
Nit: Don't use `behavior` twice in the same sentence. Maybe this instead?: "This is identical to the behavior of pipes in the shell."

Reply:
@Trott I definitely see what you mean, maybe something like "...this can cause the child process to block waiting for the pipe buffer to accept more data." would be better? Paraphrased from the the Python [docs for subprocess](https://docs.python.org/3/library/subprocess.html#popen-objects) to describe a similar situation

Reply:
I like that. (I also don't fell *that* strongly about "hang" so if you or others prefer to keep it that way, you can ignore my comment.)